scrapy/exporters.py

```diff
@@ -238,9 +244,12 @@ def export_item(self, item):
             self._headers_not_written = False
             self._write_headers_and_set_fields_to_export(item)
 
-        fields = self._get_serialized_fields(item, default_value='',
+        serialized_fields = self._get_serialized_fields(item, default_value='',
                                              include_empty=True)
-        values = list(self._build_row(x for _, x in fields))
+        fields = self._get_fields_from_item(item)
+        if (self.fields_to_export == None or set(fields) != set(self.fields_to_export)) and not self._data_loss_already_warned:
+            logger.warning("Data loss detected when exporting items -- This message won't be shown in further items")
```

Comment:
```suggestion
            logger.warning("Data loss detected when exporting items -- This message won't be shown for further items") 
```
I believe this sounds better, although I'm no native English speaker.

Reply:
Yes, it does sound better!